daemon/src/maker_inc_connections.rs

```diff
@@ -106,39 +129,32 @@ impl Actor {
             .map_ok(move |msg| FromTaker { taker_id, msg })
             .map(forward_only_ok::Message);
 
-        let (out_msg_actor_address, mut out_msg_actor_context) = xtra::Context::new(None);
-
-        let (forward_to_cfd, forward_to_cfd_fut) =
+        let (forward_only_ok, forward_only_ok_fut) =
             forward_only_ok::Actor::new(self.taker_msg_channel.clone_channel())
                 .create(None)
                 .run();
-        self.tasks.add(forward_to_cfd_fut);
-
-        // only allow outgoing messages while we are successfully reading incoming ones
-        let heartbeat_interval = self.heartbeat_interval;
-        let taker_disconnected_channel = self.taker_disconnected_channel.clone_channel();
-        self.tasks.add(async move {
-            let mut actor = send_to_socket::Actor::new(write, transport_state.clone());
-
-            let _heartbeat_handle = out_msg_actor_context
-                .notify_interval(heartbeat_interval, || wire::MakerToTaker::Heartbeat)
-                .expect("actor not to shutdown")
-                .spawn_with_handle();
+        taker_tasks.add(forward_only_ok_fut);
+        taker_tasks.add(forward_only_ok.clone().attach_stream(read));
 
-            out_msg_actor_context
-                .handle_while(&mut actor, forward_to_cfd.attach_stream(read))
-                .await;
+        let (out_msg, mut out_msg_actor_context) = xtra::Context::new(None);
+        let send_to_socket_actor = send_to_socket::Actor::new(write, transport_state.clone());
 
-            tracing::error!("Closing connection to taker {}", taker_id);
-            let _ = taker_disconnected_channel
-                .send(maker_cfd::TakerDisconnected { id: taker_id })
-                .await;
-
-            actor.shutdown().await;
-        });
-
-        self.write_connections
-            .insert(taker_id, out_msg_actor_address);
+        let heartbeat_interval = self.heartbeat_interval;
+        let heartbeat_fut = out_msg_actor_context
+            .notify_interval(heartbeat_interval, || wire::MakerToTaker::Heartbeat)
+            .expect("actor not to shutdown");
+        taker_tasks.add(heartbeat_fut);
+
+        out_msg_actor_context.run(send_to_socket_actor).await;
```

Comment:
This is blocking but must not block the current future. Needs to be a separate task.

Reply:
This was the bug, thank you!